File: src/main/java/seremis/geninfusion/soul/entity/ai/EntityAIRestrictSunCustom.java
package seremis.geninfusion.soul.entity.ai;

import net.minecraft.entity.EntityLiving;
import net.minecraft.entity.ai.EntityAIBase;
import seremis.geninfusion.api.soul.IEntitySoulCustom;

public class EntityAIRestrictSunCustom extends EntityAIBase {

    public IEntitySoulCustom entity;
    public EntityLiving living;

    public EntityAIRestrictSunCustom(IEntitySoulCustom entity) {
        this.entity = entity;
        this.living = (EntityLiving) entity;
    }

    public boolean shouldExecute() {
        return this.entity.getWorld().isDaytime();
    }

    public void startExecuting() {
        this.living.getNavigator().setAvoidSun(true);
    }

    public void resetTask() {
        this.living.getNavigator().setAvoidSun(false);
    }
}

File: src/main/scala/seremis/geninfusion/soul/entity/ai/EntityAIRestrictSunCustom.scala
package seremis.geninfusion.soul.entity.ai

import net.minecraft.entity.EntityLiving
import net.minecraft.entity.ai.EntityAIBase
import seremis.geninfusion.api.soul.IEntitySoulCustom

class EntityAIRestrictSunCustom(var entity: IEntitySoulCustom) extends EntityAIBase {

    var living: EntityLiving = entity.asInstanceOf[EntityLiving]

    override def shouldExecute(): Boolean = entity.getWorld_I.isDaytime

    override def startExecuting() {
        this.living.getNavigator.setAvoidSun(true)
    }

    override def resetTask() {
        this.living.getNavigator.setAvoidSun(false)
    }
}